File: AgoraChat/Code/Custom/View/ACDNormalNavigationView.m
//
//  ACDNormalNavigationView.m
//  AgoraChat
//
//  Created by 杜洁鹏 on 2022/7/7.
//  Copyright © 2022 easemob. All rights reserved.
//

#import "ACDNormalNavigationView.h"

@implementation ACDNormalNavigationView


- (void)prepare {
    [self addSubview:self.leftLabel];
    [self addSubview:self.leftButton];
    [self addSubview:self.rightButton];
}

- (void)placeSubViews {
    [self.leftButton mas_makeConstraints:^(MASConstraintMaker *make) {
        make.top.equalTo(self).offset(kAgroaPadding * 2.0);
        make.left.equalTo(self).offset(kAgroaPadding);
        make.width.equalTo(@40.0);
        make.bottom.equalTo(self).offset(-5.0);
    }];
    
    [self.leftLabel mas_makeConstraints:^(MASConstraintMaker *make) {
        make.centerY.equalTo(self.leftButton);
        make.left.equalTo(self.leftButton.mas_right).offset(kAgroaPadding);
    }];
    
    [self.rightButton mas_makeConstraints:^(MASConstraintMaker *make) {
        make.top.equalTo(self).offset(kAgroaPadding * 2.0);
        make.right.equalTo(self.mas_right).offset(-kNavigationRightPadding);
        make.bottom.equalTo(self).offset(-5.0);
    }];
}

#pragma mark action


- (void)leftButtonAction {
    if (self.leftButtonBlock) {
        self.leftButtonBlock();
    }
}

- (void)rightButtonAction {
    if (self.rightButtonBlock) {
        self.rightButtonBlock();
    }
}

#pragma mark getter and setter
- (UIButton *)leftButton {
    if (_leftButton == nil) {
        _leftButton = [[UIButton alloc] initWithFrame:CGRectMake(0, 0, 8, 15)];
        _leftButton.contentMode = UIViewContentModeScaleAspectFill;
        [_leftButton setImage:ImageWithName(@"black_goBack") forState:UIControlStateNormal];
        [_leftButton addTarget:self action:@selector(leftButtonAction) forControlEvents:UIControlEventTouchUpInside];
    }
    return _leftButton;
}


- (UILabel *)leftLabel {
    if (_leftLabel == nil) {
        _leftLabel = UILabel.new;
        _leftLabel.textColor = TextLabelBlackColor;
        _leftLabel.textAlignment = NSTextAlignmentLeft;
        _leftLabel.font = BFont(14.0f);
        _leftLabel.text = @"leftLabel";

    }
    return _leftLabel;
}

- (UIButton *)rightButton {
    if (!_rightButton) {
        _rightButton = [[UIButton alloc] initWithFrame:CGRectMake(0, 0, 8, 15)];
        _rightButton.contentMode = UIViewContentModeScaleAspectFill;
        [_rightButton addTarget:self action:@selector(rightButtonAction) forControlEvents:UIControlEventTouchUpInside];
    }
    return _rightButton;
}

@end

File: AgoraChat-Swift/AgoraChat-Swift/code/Custom/View/ACDNormalNavigationView.swift
//
//  ACDNormalNavigationView.swift
//  AgoraChat-Swift
//
//  Created by 冯钊 on 2022/10/31.
//

import UIKit
import SnapKit

class ACDNormalNavigationView: UIView {

    let leftButton = UIButton(frame: CGRect(x: 0, y: 0, width: 8, height: 15))
    let leftLabel = UILabel()
    let rightButton = UIButton(frame: CGRect(x: 0, y: 0, width: 8, height: 15))
    
    var leftButtonHandle: (() -> Void)?
    var rightButtonHandle: (() -> Void)?

    override init(frame: CGRect) {
        super.init(frame: frame)
        self.selfInit()
    }
    
    required init?(coder: NSCoder) {
        super.init(coder: coder)
        self.selfInit()
    }
    
    private func selfInit() {
        self.leftButton.contentMode = .scaleAspectFill;
        self.leftButton.setImage(UIImage(named: "black_goBack"), for: .normal)
        self.leftButton.addTarget(self, action: #selector(leftButtonAction), for: .touchUpInside)
        self.addSubview(self.leftButton)
        
        self.leftLabel.textColor = .black
        self.leftLabel.textAlignment = .left
        self.leftLabel.font = UIFont.systemFont(ofSize: 14)
        self.addSubview(self.leftLabel)
        
        self.rightButton.contentMode = .scaleAspectFill
        self.rightButton.addTarget(self, action: #selector(rightButtonAction), for: .touchUpInside)
        self.addSubview(self.rightButton)
        
        self.leftButton.snp.makeConstraints { make in
            make.top.equalTo(20)
            make.left.equalTo(10)
            make.width.equalTo(40.0);
            make.bottom.equalTo(-5.0)
        }
        
        self.leftLabel.snp.makeConstraints { make in
            make.centerY.equalTo(self.leftButton)
            make.left.equalTo(self.leftButton.snp.right).offset(10)
        }
        
        self.rightButton.snp.makeConstraints { make in
            make.centerY.equalTo(self.leftButton)
            make.right.equalTo(-16)
        }
    }
    
    @objc private func leftButtonAction() {
        self.leftButtonHandle?()
    }
    
    @objc private func rightButtonAction() {
        self.rightButtonHandle?()
    }
    
}
